tests/manage/pv_services/test_pvc_evict_ceph_clients.py

```diff
@@ -0,0 +1,100 @@
+from ocs_ci.framework.testlib import bugzilla
+from ocs_ci.ocs import constants
+from ocs_ci.ocs.node import get_worker_nodes
+from ocs_ci.helpers import helpers
+from ocs_ci.ocs.resources.pod import get_ceph_tools_pod
+
+logger = logging.getLogger(__name__)
+
+
+@bugzilla("1901499")
+class TestPvcEvictCephClients:
+    """
+    Test the ceph-fs PVC (RWX) mount with the following scenarios involving evict ceph-fs clients:
+    1. When two pods tries to mount the same PVC and scheduled on same node
+    2. When two pods tries to mount the same PVC and scheduled on different node
+    """
+
+    @pytest.mark.parametrize(
+        argnames=["interface", "node"],
+        argvalues=[
+            pytest.param(
+                *[constants.CEPHFILESYSTEM],
+                "same",
+            ),
+            pytest.param(
+                *[constants.CEPHFILESYSTEM],
+                "different",
+            ),
+        ],
+    )
+    def test_pvc_evict_ceph_clients(
+        self, interface, node, pvc_factory, teardown_factory
+    ):
+        worker_nodes = get_worker_nodes()
+
+        # create a RWX PVC
+        pvc_obj = pvc_factory(
+            interface=interface,
+            access_mode=constants.ACCESS_MODE_RWX,
+            status=constants.STATUS_BOUND,
+        )
+
+        # create a pod in particular node
+        selected_node = random.choice(worker_nodes)
+        logger.info(f"creating first pod on node {selected_node}")
+        first_pod_obj = helpers.create_pod(
+            interface_type=interface,
+            pvc_name=pvc_obj.name,
+            namespace=pvc_obj.namespace,
+            node_name=selected_node,
+            pod_dict_path=constants.NGINX_POD_YAML,
+        )
+        teardown_factory(first_pod_obj)
+        helpers.wait_for_resource_state(
+            resource=first_pod_obj, state=constants.STATUS_RUNNING, timeout=120
+        )
+
+        # evict ceph-fs clients
+        logger.info("Evicting ceph-fs clients!")
+        ceph_tools_pod_obj = get_ceph_tools_pod()
+        cmd_ls = "ceph tell mds.0 client ls"
+        output_ls_cmd = json.loads(
+            ceph_tools_pod_obj.exec_sh_cmd_on_pod(command=cmd_ls)
+        )
+        ids = list()
+        ids.extend([output_ls_cmd[0]["id"], output_ls_cmd[1]["id"]])
+        logger.info(f"Client Ids to evict: {ids}")
+        try:
+            for id in ids:
+                cmd_evict = f"ceph tell mds.0 client evict id={id}"
+                ceph_tools_pod_obj.exec_sh_cmd_on_pod(command=cmd_evict)
+        except CommandFailed as e:
+            assert False, logger.error(f"[Error] Client eviction failed: {e}")
+        else:
+            logger.info("Clients are evicted successfully!")
+
+        # running second pod on same/different node
+        logger.info(f"Running second pod on {node} node")
+        if node == "different":
+            worker_nodes.remove(selected_node)
+            selected_node = random.choice(worker_nodes)
```

Comment:
Does this always ensure that the randomly selected node is not the same as the node where the pod was running initially?

Reply:
Yes, I removed the earlier node from the list here `            worker_nodes.remove(selected_node)`